src/pytorch_lightning/strategies/ipu.py

```diff
@@ -272,6 +245,20 @@ def to_tensor(x):
         args = apply_to_collection(args, dtype=(int, float), function=to_tensor)
         return args
 
+    def batch_to_device(self, batch: Any, device: Optional[torch.device] = None, dataloader_idx: int = 0) -> Any:
+        # This override is necessary because the cast must occur before the data
+        # is moved to the device to prevent wasteful host->device copies.
+        if self.precision_plugin.precision in (PrecisionType.MIXED, PrecisionType.HALF):
+
+            def to_half(data: Tensor) -> Tensor:
+                return data.half()
+
+            batch = apply_to_collection(batch, (FloatTensor, torch.cuda.FloatTensor), function=to_half)
+        # We don't call `super().batch_to_device` because `data.to(device)` is not
+        # currently necessary for IPUs. The movement of data from host<->IPU is
+        # currently handled by PopTorch.
```

Comment:
if that's the case, we need to do a follow-up for users to let them know that some of the hooks won't work with IPUs, like `transfer_batch_to_device`, ...

Reply:
Is this something you could do? I don't know the codebase well enough to know all the implications.